ocaml/typing/typedecl_variance.ml

```diff
@@ -192,7 +192,7 @@ let compute_variance_type env ~check (required, loc) decl tyl =
                                                         (c,n,i)))))
       params required;
     (* Check propagation from constrained parameters *)
-    let args = Btype.newgenty (Ttuple params) in
+    let args = Btype.newgenty (Ttuple (List.map (fun t -> None, t) params)) in
     let fvl = Ctype.free_variables args in
```

Comment:
Maybe leave a CR to check whether it's worth adding a proper `free_variables_list`.